.github/workflows/docker-unified.yml

```diff
@@ -71,6 +71,9 @@ jobs:
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
+      - name: Pre-build artifacts for docker image
+        run: |
+          ./gradlew :metadata-service:war:build -x test --parallel
```

Comment:
I like where this is going, is it possible to add a gradle task like publishToDockerBuild or something like publishToMavenLocal for each module which would build and then copy the artifact into docker? This would make it possible to mirror this same process with a gradle command for doing this locally. I can see something like ./gradlew docker (building these locally)

Reply:
You mean something like?:

```
DockerTask extends gradle.CacheableTask {
// Runs equivalent of ./gradlew <BUILD_ARTIFACTS> for all necessary modules
...
// Runs docker build --tag local from within gradle code for all docker builds
...
}
```

I think the main effort there is if gradle has docker integrations/support for directly running CLI commands through code. If it has either of those options should be possible to do this within a gradle task. Might be easier in a bash script though

Reply:
I have an idea here. Let me try it.